systemtest/src/main/java/io/strimzi/systemtest/resources/operator/OlmResource.java

```diff
@@ -6,71 +6,280 @@
+    public static void obtainInstallPlanName() {
+        String installPlansPureString = cmdKubeClient().exec("get", "installplan").out();
+        String[] installPlansLines = installPlansPureString.split("\n");
+
+        for (String line : installPlansLines) {
+            // line NAME  CSV  APPROVAL   APPROVED
+            String[] wholeLine = line.split(" ");
+
+            // name
+            if (wholeLine[0].startsWith("install-")) {
+
+                // if is not already applied add to closed map
+                if (!CLOSED_MAP_INSTALL_PLAN.containsKey(wholeLine[0])) {
+                    LOGGER.info("CLOSED_MAP_INSTALL_PLAN does not contain {} install plan so this is not used and will " +
+                        "be in the following upgrade.", wholeLine[0]);
+                    CLOSED_MAP_INSTALL_PLAN.put(wholeLine[0], Boolean.FALSE);
+                }
+            }
+        }
+        if (!(CLOSED_MAP_INSTALL_PLAN.keySet().size() > 0)) {
+            throw new RuntimeException("No install plans located in namespace:" + cmdKubeClient().namespace());
+        }
+    }
+
+    /**
+     * Get specific version of cluster operator with prefix name in format: 'strimzi-cluster-operator.v0.18.0'
+     * @return version with prefix name
+     */
+    public static String getClusterOperatorVersion() {
+        String installPlansPureString = cmdKubeClient().exec("get", "installplan").out();
+        String[] installPlansLines = installPlansPureString.split("\n");
+
+        for (String line : installPlansLines) {
+            // line = NAME   CSV   APPROVAL   APPROVED
+            String[] wholeLine = line.split("   ");
+
+            // non-used install plan
+            if (wholeLine[0].equals(getNonUsedInstallPlan())) {
+                return wholeLine[1];
+            }
+        }
+        throw new RuntimeException("Version was not found in the install plan.");
+    }
+
+    public static boolean isUpgradeable() {
+        return !getNonUsedInstallPlan().equals(NO_MORE_NON_USED_INSTALL_PLANS);
+    }
+
+    public static String getNonUsedInstallPlan() {
+        String[] nonUsedInstallPlan = new String[1];
+
+        for (Map.Entry<String, Boolean> entry : CLOSED_MAP_INSTALL_PLAN.entrySet()) {
+            // if value is FALSE we are gonna use it = non-used install plan
+            if (!entry.getValue()) {
+                nonUsedInstallPlan[0] = entry.getKey();
+                break;
+            }
+            nonUsedInstallPlan[0] = NO_MORE_NON_USED_INSTALL_PLANS;
+        }
+
+        LOGGER.info("Non-used install plan is {}", nonUsedInstallPlan[0]);
+        return nonUsedInstallPlan[0];
+    }
+
+    /**
+     * Patches specific non used install plan, which will approve installation. Only for manual installation strategy.
+     * Also updates closedMapInstallPlan map and set specific install plan to true.
+     */
+    private static void approveInstallation() {
+        String nonUsedInstallPlan = getNonUsedInstallPlan();
```

Comment:
This is a little bit confusing. You want to patch and approve non used install plan. Maybe better naming?